Gruvest-Django/mysite/myapp/views.py

```diff
@@ -33,13 +34,72 @@ def form_valid(self, form):
     def get_success_url(self):
         return '/'
 
+class CommentCreator(CreateView):
+    # the associated html template
+    template_name = "post_comment.html"
+    # specified model which this object creates
+    model = models.CommentModel
+    # specified fields to be entered by user
+    fields = [
+        "comment"
+    ]
+    form = forms.PostCommentForm
+    # upon creation, stay on current page (which is main since PostModel redirects to main)
+    success_url = '/'
+    # error checking form
+    def form_valid(self, form):
+        form.instance.post_id = self.kwargs['pk']
+        return super().form_valid(form) # call parent object method
+    
+    def get_success_url(self):
+        return '/'
+
+class AddFunds(UpdateView):
+    # the associated html template
+    template_name = "add_funds.html"
+    # specified model which this object creates
+    model = models.CatcherModel
+    # specified fields to be entered by user
+    fields = [
+        'funds'
+    ]
+    form = forms.AddFundsForm
+    # upon creation, stay on current page (which is main since PostModel redirects to main)
+    success_url = '/'
+
+    def post(self, request, *args, **kwargs):
+        form = self.form(request.POST)
+        if form.is_valid():
+            request.user.catchermodel.funds += int(form['funds'].value())
+            request.user.catchermodel.save()
+        return HttpResponseRedirect(reverse("main"))
+    # error checking form
+    def form_valid(self, form):
+        print(form.cleaned_data)
+        return super().form_valid(form) # call parent object method
+    
+    def get_success_url(self):
+        return '/'
+
 '''
 PitchDetail inherits from DetailView
     is a Retrieve operation
 '''
 class PitchDetail(DetailView):
     template_name = "view_pitch.html"
     model = models.PostModel
+'''
```

Comment:
I know this is commented but can you explain what this is for?

Reply:
The purpose of this function is the check if the user has enough funds to purchase the ability to view the full post but I have gotten it to fully work yet.